lib/valkyrie/persistence/fedora/query_service.rb

```diff
@@ -139,14 +137,14 @@ def find_inverse_references_by_unordered(resource:, property:)
         property_uri = adapter.schema.predicate_for(property: property, resource: nil)
         ids = content.graph.query([nil, property_uri, adapter.id_to_uri(resource.id)]).map(&:subject).map { |x| x.to_s.gsub(/#.*/, '') }.map { |x| adapter.uri_to_id(x) }
         ids.uniq!
-        ids.lazy.map { |id| find_by(id: id) }
+        ids
       end
 
       def find_inverse_references_by_ordered(resource:, property:)
         content = content_with_inbound(id: resource.id)
         ids = content.graph.query([nil, ::RDF::Vocab::ORE.proxyFor, adapter.id_to_uri(resource.id)]).map(&:subject).map { |x| x.to_s.gsub(/#.*/, '') }.map { |x| adapter.uri_to_id(x) }
         ids.uniq!
-        ids.lazy.map { |id| find_by(id: id) }.select { |o| o[property].include?(resource.id) }
```

Comment:
I think we still need this check. If a resource has several ordered properties, this is the only way we would know if the one we care about is pointing at the resource, instead of another one.